### PR TITLE
add me.proton.Pass

### PR DESCRIPTION
**Pull request by proletarius101**
### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly: password manager created by the scientists who met at CERN behind Proton Mail, the world’s largest encrypted email provider. Proton Pass is open source, end-to-end encrypted, and protected by Swiss privacy laws.
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://protonmail.uservoice.com/forums/953584-proton-pass/suggestions/48539555-publish-proton-pass-to-flathub
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: me.proton.Pass.metainfo.xml
Comment (hfiguiere) [inline me.proton.Pass.metainfo.xml:1]: this should be submitted upstream

File: me.proton.Pass.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+      - install -Dm644 proton-pass.svg "${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg"
+      - install -Dm644 usr/share/applications/proton-pass.desktop "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"
+      - desktop-file-edit --set-key=Exec --set-value='start-proton-pass %U' --set-icon=${FLATPAK_ID}
+        "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"
+    sources:
+      - type: file
+        dest-filename: ProtonPass.deb
+        only-arches: [x86_64]
+        url: https://proton.me/download/PassDesktop/linux/x64/ProtonPass_1.17.5.deb
```
Comment (hfiguiere) [inline me.proton.Pass.yml:46]: so it's build an aarch64 package still.

File: me.proton.Pass.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+            if [[ "${USE_WAYLAND}" -eq 1 && "${XDG_SESSION_TYPE}" == "wayland" ]]; then
+                exec zypak-wrapper "/app/proton-pass/Proton Pass" --ozone-platform-hint=wayland --enable-features=WaylandWindowDecorations --gtk-version=4 $@ || EXIT_CODE=$?
+                # Fall back to x11 if failed to launch under Wayland. Otherwise, exit normally
+                [[ "${EXIT_CODE}" -ne 133 ]] && exit "${EXIT_CODE}"
+            fi
+
+            exec zypak-wrapper "/app/proton-pass/Proton Pass" $@
+
+  - name: metainfo
```
Comment (hfiguiere) [inline me.proton.Pass.yml:96]: this should be part of the main module.

File: me.proton.Pass.yml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      #     project-id: 146681
+      #     url-template: https://github.com/standardnotes/app/releases/download/%40standardnotes%2Fdesktop%40$version/standard-notes-$version-linux-arm64.deb
+      #     stable-only: true
+
+      - type: file
+        dest-filename: proton-pass.svg
+        url: https://raw.githubusercontent.com/ProtonMail/WebClients/d1a23fc2c4ab1e1632118baaee6f2531022ae4cf/applications/pass-desktop/assets/logo.svg
+        sha256: 485abebe08af491753b9060c539f012c2c0a8d20a1e559d42f623c281ed691b9
+
+  - name: start-script
```
Comment (hfiguiere) [inline me.proton.Pass.yml:69]: this should be part of the main module.

**Comment by hfiguiere** [comment]
as usual, building from source would be better.

**Comment by hfiguiere** [comment]
> > as usual, building from source would be better.
> 
> Unfortunately, the `flatpak-node-generator` is unable to properly parse https://github.com/ProtonMail/WebClients/blob/main/yarn.lock

if only each time one complained we actually got help to fix it maybe they would be less complains.

File: me.proton.Pass.yml (hunk 1 above)
Comment (hfiguiere) [reply me.proton.Pass.yml:51]: how disappointing....

File: me.proton.Pass.yml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+base-version: *runtime-version
+command: start-proton-pass
+finish-args:
+  # camera access for moments
+  - --device=all
+  - --share=ipc
+  - --share=network
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --talk-name=org.freedesktop.Notifications
```
Comment (bbhtt) [inline me.proton.Pass.yml:15]:
```suggestion
```
Not needed with libnotify 0.8.0 and up which is in baseapp. It uses the portal.

File: me.proton.Pass.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            <li>Secure your Pass account with additional security measures: Protect all your data with
+                another layer of protection, either with TOTP or U2F/FIDO2 security keys.
+            </li>
+            <li>Get unlimited email forwards: There’s no limit on the number of emails you can have
+                forwarded from your alias to your inbox.</li>
+        </ul>
+    </description>
+
+    <url type="homepage">https://proton.me/pass</url>
+    <url type="vcs-browser">https://raw.githubusercontent.com/ProtonMail/WebClients</url>
```
Comment (bbhtt) [inline me.proton.Pass.metainfo.xml:87]:
```suggestion
    <url type="vcs-browser">https://github.com/ProtonMail/WebClients</url>
```

File: me.proton.Pass.metainfo.xml
Comment (bbhtt) [inline me.proton.Pass.metainfo.xml:1]: Can you please submit this upstream?

**Pull request closed without merging**